Check recovery key

matthew-hagemann opened the pull request:
This PR begins the work to bring the TPM & recovery key features to the security center:
- Adds a new page for disk encryption actions
- Adds checking a recovery key (currently only against the mock while we wait for the API from snapd)
- Scaffolds out some of the other functions that will be used later (did this to help my own mental model for how the data should look)

## New page:
![Screenshot From 2025-06-03 14-33-05](https://github.com/user-attachments/assets/6cdb7659-e1d3-4167-801b-3153fc4dbd81)
## Dialog when you click the button, auto focus
![Screenshot From 2025-06-03 17-58-57](https://github.com/user-attachments/assets/ae5ecf60-fa1d-4a74-be1a-de666f4f8e3f)
## Can check when you enter a recovery key
![Screenshot From 2025-06-03 18-00-07-1](https://github.com/user-attachments/assets/e46b51df-e49c-42ac-a271-7f91398acca2)
## Valid case
![Screenshot From 2025-06-03 14-33-20](https://github.com/user-attachments/assets/62b8efaf-fd74-4f68-9d8f-42e2d4255b1a)
## Invalid key case
![Screenshot From 2025-06-03 14-33-26](https://github.com/user-attachments/assets/25607bb5-8845-47f9-8ee1-b4d819091c27)

d-loose commented:
I think one major simplification that we should look at tomorrow morning is breaking down the UI state into smaller pieces - the dialog shouldn't need to handle the loading/error states related to enumerating the system volumes.

matthew-hagemann commented:
New screenshots:
- Updated copy on error message
- Fixed bottom margin
![Screenshot From 2025-06-04 16-46-02](https://github.com/user-attachments/assets/1510a6be-2166-4d14-8d6b-4d35b11ce06f)
![Screenshot From 2025-06-04 16-46-10](https://github.com/user-attachments/assets/a3785daf-aa4d-439b-8e96-94a18b17befb)

d-loose commented on packages/security_center/lib/disk_encryption/disk_encryption_page.dart:
```diff
@@ -0,0 +1,117 @@
+        ),
+        const SizedBox(height: 16),
+        Text(l10n.diskEncryptionPageStoreYourKey),
+        const SizedBox(height: 16),
```
There's a convenient `separatedBy` method in the `iterable_extensions.dart` here :)

matthew-hagemann commented:
![image](https://github.com/user-attachments/assets/de215c2b-e679-4bf1-8271-4ec9f2cd3a8e)

d-loose commented on packages/security_center/lib/services/disk_encryption_service.dart:
```diff
@@ -0,0 +1,99 @@
+part 'disk_encryption_service.freezed.dart';
+part 'disk_encryption_service.g.dart';
+
+/// Dialog state for managing the recovery key check process.
+@freezed
+sealed class CheckRecoveryKeyDialogState with _$CheckRecoveryKeyDialogState {
+  factory CheckRecoveryKeyDialogState.empty() =
+      CheckRecoveryKeyDialogStateEmpty;
+  factory CheckRecoveryKeyDialogState.input(String keyToCheck) =
+      CheckRecoveryKeyDialogStateInput;
+  factory CheckRecoveryKeyDialogState.result(bool valid) =
+      CheckRecoveryKeyDialogStateResult;
+  factory CheckRecoveryKeyDialogState.loading() =
+      CheckRecoveryKeyDialogStateLoading;
+  factory CheckRecoveryKeyDialogState.error(Exception e) =
+      CheckRecoveryKeyDialogStateError;
+}
```
a final nitpick: could you move this to the `disk_encryption_providers.dart`, as it's unrelated to the service

d-loose commented on packages/security_center/test/disk_encryption/disk_encryption_page_test.dart:
This could probably be simplified with a go-style table test, but we can save that for a future pair programming session :)

d-loose commented on packages/security_center/test/test_utils.dart:
```diff
@@ -70,14 +84,47 @@ AppPermissionsService registerMockAppPermissionsService({
-LocalSnapData registerMockLocalSnapData({
-  List<Snap> snaps = const [],
-}) {
+LocalSnapData registerMockLocalSnapData({List<Snap> snaps = const []}) {
```
some more formatting changes (they all seem to be related to trailing commas :confused: )

d-loose reviewed (approved):
Thanks, LGTM :+1:
